src/mono/mono/mini/mini-arm64.c

```diff
@@ -3933,13 +3933,15 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 
 		case OP_XZERO:
 			if (ins->klass && mono_class_value_size (ins->klass, NULL) == 8)
-				arm_neon_eor_8b (code, dreg, dreg, dreg);
+				arm_neon_movi_b (code, VREG_LOW, dreg, 0);
 			else
-				arm_neon_eor_16b (code, dreg, dreg, dreg);
+				arm_neon_movi_b (code, VREG_FULL, dreg, 0);
 			break;
 		case OP_XONES:
-			arm_neon_eor_16b (code, dreg, dreg, dreg);
-			arm_neon_not_16b (code, dreg, dreg);
+			if (ins->klass && mono_class_value_size (ins->klass, NULL) == 8)
+				arm_neon_movi_b (code, VREG_LOW, dreg, 0xff);
+			else
+				arm_neon_movi_b (code, VREG_FULL, dreg, 0xff);
```

Comment:
Could this be replaced by `arm_neon_movi_b (code, get_vector_size_macro (ins), dreg, 0xff)`?

Reply:
+1

Reply:
Absolutely!